### DISCUSSION TITLE
Roadmap: Revision History

### DISCUSSION THREAD

**Discussion by DanRibbens** [payloadcms/payload#292] [Roadmap]
### Description

Add a config option that when enabled, automatically persist the document change history over time. In the admin UI there will be controls to see previously saved versions along with timestamps and author.

### Implementation Detail

Configuration for revisions should allow the developer to enable it across all collections, globals or scoped to the document level (Collection A can have revisions on while Collection B does not).
The admin UI will need to list revisions and allow the user to list revisions of a document. The list should allow you to revert the active data back to pervious versions.

On the persistence side, the database will need to store the changes and operations need to be done to the live version. If a local API operation changes a document, we need to determine how those revisions are stored and tracked.

### Potential Breaking Changes

Investigate and plan around file uploads and plugins.

### Effort

High

**Comment by liamrohan** [comment]
Hello, is this still on target for end of Q4? Many thanks

**Reply by Jeluda** [reply]
I would like to know as well.
Thanks.

**Reply by jmikrut** [reply]
It sure is! We're working on this feature now and have a great approach that we're working toward. It will be delivered by the end of Q4 as scheduled!

**Comment by DanRibbens** [comment]
This is now released in `0.15.0`.

Thanks for the patience on the extra time it took to deliver this huge feature. That allowed us to test it within some real working projects to be sure it is ready for release. Enjoy!

**Comment by geminigeek** [comment]
Hi,

payload is just awesome, 

i have one suggestion can we have an extra field in versions to track what triggered the version creation, like where we can add a note ?

**Reply by matthijs166** [reply]
It is possible to add a new field in the collection for this and add a hook to set the field. Now the person who edited the item will show up in the versions.
